ci-operator/templates/build-azure-image-310.yaml

```diff
@@ -0,0 +1,300 @@
+          -e "openshift_azure_install_repo=\$(curl -qs https://storage.googleapis.com/origin-ci-test/releases/openshift/origin/release-3.10/.latest-rpms)" \
+          playbooks/azure/openshift-cluster/build_node_image.yml
+
+        touch /tmp/shared/image_create
+
+    # Runs an install
+    - name: buildcluster
+      image: ${LOCAL_IMAGE_BIN}
+      volumeMounts:
+      - name: shared-tmp
+        mountPath: /tmp/shared
+      - name: cluster-secrets-azure
+        mountPath: /etc/azure/credentials
+      env:
+      - name: INSTANCE_PREFIX
+        value: ${NAMESPACE}-${JOB_NAME_HASH}
+      - name: TYPE
+        value: ${CLUSTER_TYPE}
+      - name: HOME
+        value: /tmp/shared/home
+      command:
+      - /bin/bash
+      - -c
+      - |
+       #!/bin/bash
+        # error handling and sync code
+        set -euo pipefail
+
+        trap 'rc=$?; if [[ $rc -ne 0 ]]; then
+        # on failure we copy details for clean-up
+        cp -r /go/src/github.com/openshift/openshift-azure/_data /tmp/shared/ &>/dev/null
+        touch /tmp/shared/exit; fi; exit $rc' EXIT
+        trap 'kill $(jobs -p); exit 0' TERM
+
+        # Cluster creation specific configuration.
+        mkdir -p "${HOME}"
+
+
+        # wait until the setup job creates admin.kubeconfig
+        while true; do
+          if [[ ! -f  /tmp/shared/image_create ]]; then
+            sleep 15 & wait
+            continue
+          fi
+          # if we got admin kubeconfig file with failure marker, ignore. Teardown is in progress.
+          if [[ -f  /tmp/shared/exit ]]; then
+             exit 1
+          fi
+          break
+        done
+        echo "Found image create marker file."
+
+        source /etc/azure/credentials/secret
+        az login --service-principal -u ${AZURE_CLIENT_ID} -p ${AZURE_CLIENT_SECRET} --tenant ${AZURE_TENANT_ID} &>/dev/null
+        # TODO: Configure these from prow
+        export DNS_DOMAIN=osadev.cloud
+        export DNS_RESOURCEGROUP=dns
+        export DEPLOY_VERSION=v3.10
+        export RUN_SYNC_LOCAL=true
+        export DEPLOY_OS="centos7"
+        export IMAGE_RESOURCEGROUP=images
+        # Azure cli fails miserably now with az image list
+        # https://github.com/Azure/azure-cli/issues/7019
+        #export IMAGE_RESOURCENAME=$(az image list -g $IMAGE_RESOURCEGROUP -o json --query "[?starts_with(name, '${DEPLOY_OS:-rhel7}-${DEPLOY_VERSION//v}') && tags.valid=='true'].name | sort(@) | [-1]" | tr -d '"')
+        #
+        source /tmp/shared/environment
+        export IMAGE_RESOURCENAME=${IMAGE_NAME}
+        # TODO: for full e2e image needs to be built too
+        cd /go/src/github.com/openshift/openshift-azure/
+
+        ./hack/create.sh ${INSTANCE_PREFIX}
+
+        # if completed pass generated data to other stage
+        cp -r /go/src/github.com/openshift/openshift-azure/_data /tmp/shared/
+
+        touch /tmp/shared/e2e_tests
+
+    - name: tagimage
+      image: docker.io/openshift/origin-ansible:v3.10.1
```

Comment:
We should be doing this with the tag specification in ci-operator for the 3.10 branch, not here

Reply:
Agreed.  I'm not sure I understand the tag specification completely.  This made it very easy.

Reply:
Hopefully the config reference is useful https://github.com/openshift/ci-operator/blob/master/CONFIGURATION.md#tag_specification